### PR TITLE
Add support for static files to dotnet-watch

### PR DESCRIPTION
**Pull request by pranavkm**
This is largely a port of https://github.com/dotnet/aspnetcore/pull/27744 with one large difference - I added an MSBuild task to serialize the list of files from MSBuild -> dotnet-watch. In the PR this was done by relying on CLI parsing.

### CODE REVIEW

File: src/BuiltInTools/dotnet-watch/DotNetWatchContext.cs [hunk 1]
```diff
@@ -11,14 +11,16 @@ public class DotNetWatchContext
 
         public ProcessSpec ProcessSpec { get; set; }
 
-        public IFileSet FileSet { get; set; }
+        public FileSet FileSet { get; set; }
```
[the remark this answers is not in the document]
Comment (captainsafia) [reply src/BuiltInTools/dotnet-watch/DotNetWatchContext.cs:14]: Nevermind. I see that the interface was removed.

**Review by captainsafia — APPROVED** [review]

File: src/BuiltInTools/DotNetWatchTasks/FileSetSerializer.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+using Microsoft.Build.Utilities;
+using Microsoft.DotNet.Watcher.Internal;
+
+namespace DotNetWatchTasks
+{
+    public class FileSetSerializer : Task
+    {
+        public ITaskItem[] WatchFiles { get; set; }
+
+        public bool IsNetCoreApp31OrNewer { get; set; }
```
Comment (jkotalik) [inline src/BuiltInTools/DotNetWatchTasks/FileSetSerializer.cs:19]: Reminder where this is used again?

File: src/BuiltInTools/DotNetWatchTasks/FileSetSerializer.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+        public ITaskItem OutputPath { get; set; }
+
+        public string[] PackageIds { get; set; }
```
Comment (jkotalik) [inline src/BuiltInTools/DotNetWatchTasks/FileSetSerializer.cs:23]: Unused?

**Review by jkotalik — APPROVED** [review]
The FileSetSerializer is the only new par of this PR correct?

LGTM otherwise.

**Pull request merged**